packages/@glimmer/runtime/lib/helpers/get.ts

```diff
@@ -0,0 +1,131 @@
+  childRefFor,
+  childRefFromParts,
+  createComputeRef,
+  isConstRef,
+  NULL_REFERENCE,
+  valueForRef,
+} from '@glimmer/reference';
+import { internalHelper } from './internal-helper';
+
+/**
+  Dynamically look up a property on an object. The second argument to `{{get}}`
+  should have a string value, although it can be bound.
+
+  For example, these two usages are equivalent:
+
+  ```app/components/developer-detail.js
+  import Component from '@glimmer/component';
+  import { tracked } from '@glimmer/tracking';
+
+  export default class extends Component {
+    @tracked developer = {
+      name: "Sandi Metz",
+      language: "Ruby"
+    }
+  }
+  ```
+
+  ```handlebars
+  {{this.developer.name}}
+  {{get this.developer "name"}}
+  ```
+
+  If there were several facts about a person, the `{{get}}` helper can dynamically
+  pick one:
+
+  ```app/templates/application.hbs
+  <DeveloperDetail @factName="language" />
+  ```
+
+  ```handlebars
+  {{get this.developer @factName}}
+  ```
+
+  For a more complex example, this template would allow the user to switch
+  between showing the user's height and weight with a click:
+
+  ```app/components/developer-detail.js
+  import Component from '@glimmer/component';
+  import { tracked } from '@glimmer/tracking';
+
+  export default class extends Component {
+    @tracked developer = {
+      name: "Sandi Metz",
+      language: "Ruby"
+    }
+
+    @tracked currentFact = 'name'
+
+    @action
+    showFact(fact) {
+      this.currentFact = fact;
+    }
+  }
+  ```
+
+  ```app/components/developer-detail.js
+  {{get this.developer this.currentFact}}
+
+  <button {{on 'click' (fn this.showFact "name")}}>Show name</button>
+  <button {{on 'click' (fn this.showFact "language")}}>Show language</button>
+  ```
+
+  The `{{get}}` helper can also respect mutable values itself. For example:
+
+  ```app/components/developer-detail.js
+  <Input @value={{mut (get this.person this.currentFact)}} />
+
+  <button {{on 'click' (fn this.showFact "name")}}>Show name</button>
+  <button {{on 'click' (fn this.showFact "language")}}>Show language</button>
+  ```
+
+  Would allow the user to swap what fact is being displayed, and also edit
+  that fact via a two-way mutable binding.
+
+  @public
+  @method get
+ */
+export default internalHelper((args: VMArguments) => {
+  let sourceRef = args.positional.at(0);
+  let pathRef = args.positional.at(1);
+
+  if (isConstRef(pathRef)) {
+    // Since the path is constant, we can create a normal chain of property
+    // references. The source reference will update like normal, and all of the
+    // child references will update accordingly.
+    let path = valueForRef(pathRef);
+
+    if (path === undefined || path === null || path === '') {
+      return NULL_REFERENCE;
+    } else if (typeof path === 'string' && path.indexOf('.') > -1) {
+      return childRefFromParts(sourceRef, path.split('.'));
+    } else {
+      return childRefFor(sourceRef, String(path));
+    }
```

Comment:
It seems vaguely odd to hard code the details of `getPath` here. In the else case just below we directly use `getPath` which can be customized by the host, but in this branch we manually encode _one possible_ implementation of `getPath`. IMHO, either the host is allowed to customize `getPath` behavior or it isn't. If it _can_ this truthy branch shouldn't make assumptions about how `getPath` works.

The concrete concern here is that I'd vaguely prefer to _avoid_ making Glimmer.js have the same `.` means path behavior (and therefore preventing users from being able to use `get` to access properties with `.` in them). Things like https://github.com/emberjs/ember.js/issues/17529 https://github.com/emberjs/ember.js/issues/17927 are discussing, but even if we don't actually want to support that this branch represents a "leaky abstraction" (and likely suggests that we don't need `getPath` on global context at all).

Reply:
That seems reasonable, though I do think this could be an issue for cross-compatibility (annoyingly, as I agree). We can probably remove this branch and just make the default the dynamic path, I don't think it would be likely to have an impact on perf